rust/src/types.rs

```diff
@@ -938,6 +938,128 @@ impl Type {
         let mut name = QualifiedName::from(name);
         unsafe { BnStr::from_raw(BNGenerateAutoDemangledTypeId(&mut name.0)) }
     }
+
+    pub fn demangle_gnu3<Mangled: BnStrCompatible>(
```

Comment:
I'm just going to ask that you move these to a new `demangle.rs` to match the structure of the Python/C++ APIs, please.